tests/upgrade/core-ahead/task.yaml

```diff
@@ -0,0 +1,42 @@
+summary: Check that upgrading ubuntu-core from edge snapd still works
+systems: [-ubuntu-core-16-64]
+environment:
+    CONF: /etc/systemd/system/snapd.service.d/local.conf
+prepare: |
+    cp $CONF $CONF.save
+    # ensure we use the re-exec feature
+    sed -i 's/SNAP_REEXEC=0/SNAP_REEXEC=1/' $CONF
+restore: |
+    rm -f /var/tmp/myevil.txt
+    cp $CONF.save $CONF
+execute: |
+    echo "Install current distro version..."
+    apt install -y snapd
+
+    echo "Install a test snap with it"
+    snap install test-snapd-tools
+
+    echo "Sanity check install"
+    test-snapd-tools.echo Hello | grep Hello
+    test-snapd-tools.env | grep SNAP_NAME=test-snapd-tools
+
+    echo "Do upgrade ubuntu-core to the edge channel"
+    sudo snap refresh --edge ubuntu-core
```

Comment:
likely here we would want at some point to build a ubuntu-core with the snapd/snap from the branch I suppose